homeassistant/config_entries.py

```diff
@@ -730,20 +730,26 @@ async def async_post_init(
     ) -> None:
         """After a flow is initialised trigger new flow notifications."""
         source = flow.context["source"]
+        persistent_notification = self.hass.components.persistent_notification
 
         # Create notification.
         if source in DISCOVERY_SOURCES:
             self.hass.bus.async_fire(EVENT_FLOW_DISCOVERED)
-            self.hass.components.persistent_notification.async_create(
-                title="New devices discovered",
-                message=(
-                    "We have discovered new devices on your network. "
-                    "[Check it out](/config/integrations)."
-                ),
-                notification_id=DISCOVERY_NOTIFICATION_ID,
-            )
-        elif source == SOURCE_REAUTH:
-            self.hass.components.persistent_notification.async_create(
+            if not persistent_notification.async_exists(DISCOVERY_NOTIFICATION_ID):
+                persistent_notification.async_create(
```

Comment:
We could also update `async_create` to not call a service but instead just do the action. Doesn't have to be routed via a service.

Reply:
Done in https://github.com/home-assistant/core/pull/57157